spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/cassandra/CassandraDriverHealthIndicatorTests.java

```diff
@@ -45,30 +55,86 @@ void createWhenCqlSessionIsNullShouldThrowException() {
 		assertThatIllegalArgumentException().isThrownBy(() -> new CassandraDriverHealthIndicator(null));
 	}
 
+	@ParameterizedTest
+	@MethodSource
+	void reportCassandraHealthCheck(Map<UUID, Node> nodes, Status expectedStatus) {
```

Comment:
I am not a big fan of `@ParameterizedTest` for this and this looks inconsistent with all the other test. We have rather dedicated test where the expectations are described, potentially sharing a private method that setup the infrastructure. 

Can you please give that a try? If you need inspiration, the other health indicator tests should help.